src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -6,20 +6,34 @@
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
-    private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final List<E> list = new ArrayList<>();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
         // write your code here
+        try {
+            lock.writeLock().lock();
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
```

Comment:
Consider adding a null check for the 'element' parameter to prevent adding null values to the list, which could lead to NullPointerExceptions when attempting to use the elements of the list later.